src/test/unit/mcmc/hmc/hamiltonians/dense_e_metric_test.cpp

```diff
@@ -17,33 +18,55 @@ TEST(McmcDenseEMetric, sample_p) {
   q(0) = 5;
   q(1) = 1;
 
+  Eigen::MatrixXd  m(2,2);
+  m(0,0) = 3.0;
+  m(1,0) = -2.0;
+  m(0,1) = -2.0;
+  m(1,1) = 4.0;
+
+  Eigen::MatrixXd  m_inv(2,2);
+  double det_inv = 1.0/(3.0*4.0 - (-2.0)*(-2.0));
+  m_inv(0,0) = m(1,1)*det_inv;
+  m_inv(1,0) = -1.0*m(1,0)*det_inv;
+  m_inv(0,1) = -1.0*m(0,1)*det_inv;
+  m_inv(1,1) = m(0,0)*det_inv;
+
   stan::mcmc::mock_model model(q.size());
 
   stan::mcmc::dense_e_metric<stan::mcmc::mock_model, rng_t> metric(model);
   stan::mcmc::dense_e_point z(q.size());
+  z.set_metric(m_inv);
 
   int n_samples = 1000;
-  double m = 0;
-  double m2 = 0;
+
+  Eigen::MatrixXd sample_cov(2,2);
+  sample_cov(0,0) = 0.0;
+  sample_cov(0,1) = 0.0;
+  sample_cov(1,0) = 0.0;
+  sample_cov(1,1) = 0.0;
 
   for (int i = 0; i < n_samples; ++i) {
     metric.sample_p(z, base_rng);
-    double tau = metric.tau(z);
-
-    double delta = tau - m;
-    m += delta / static_cast<double>(i + 1);
-    m2 += delta * (tau - m);
+    sample_cov(0,0) += z.p[0]*z.p[0]/(n_samples+0.0);
+    sample_cov(0,1) += z.p[0]*z.p[1]/(n_samples+0.0);
+    sample_cov(1,0) += z.p[1]*z.p[0]/(n_samples+0.0);
+    sample_cov(1,1) += z.p[1]*z.p[1]/(n_samples+0.0);
   }
 
-  double var = m2 / (n_samples + 1.0);
-
-  // Mean within 5sigma of expected value (d / 2)
-  EXPECT_TRUE(std::fabs(m   - 0.5 * q.size()) < 5.0 * sqrt(var));
-
-  // Variance within 10% of expected value (d / 2)
-  EXPECT_TRUE(std::fabs(var - 0.5 * q.size()) < 0.1 * q.size());
+  Eigen::MatrixXd var(2,2);
+  var(0,0) = 2*m(0,0);
```

Comment:
The autoformater might pick this up, but the style standard has spaces after commas and around operators, so `var(0, 0) = 2 * m(0, 0);`

Reply:
I am never happier than when I'm adding spaces to things!